### PR TITLE
Fix #4240

### LINKED ISSUE #4240: Images are unviewable but rendered
Under Settings -&gt; Attachments -&gt; Show Attached Thumbnails in Posts -&gt; Full Size Image

If the user can't download images these are rendered anyways.

If the user can't download the full image it might be better to render the thumbnail.

Original thread: [Images are unviewable but rendered](https://community.mybb.com/thread-230845.html)

### PR DESCRIPTION
**Pull request by Sama34**
Resolves #4240

### CODE REVIEW

**Comment by Sama34** [comment]
@andrewjs18

File: install/resources/settings.xml [hunk 1]
```diff
@@ -1537,6 +1537,14 @@ min=1]]></optionscode>
 			<settingvalue><![CDATA[96]]></settingvalue>
 			<isdefault>1</isdefault>
 		</setting>
+		<setting name="attachimagestock">
+			<title>Stock Image Path</title>
+			<description><![CDATA[Select an stock image to use for image attachments when users have no permission to download.]]></description>
```
Comment (Xazin) [inline install/resources/settings.xml]: Should be
`Select a stock image ...`

**Review by Xazin — APPROVED** [review]
Worked for me. Would have loved some settings for setting the width and height of the stock image, as it just takes the actual size of the image.

Looks good!

**Comment by Eldenroot** [comment]
Stock... isnt better to use "Default" or something like fallback"? 

About width and height - it is defined with your image itself.

File: install/resources/mybb_theme.xml [hunk 2]
```diff
@@ -9680,11 +9680,11 @@ if(use_xmlhttprequest == "1")
 		<template name="postbit_attachments_images" version="120"><![CDATA[<span class="smalltext"><strong>{$lang->postbit_attachments_images}</strong></span><br />
 {$post['imagelist']}
 <br />]]></template>
-		<template name="postbit_attachments_images_image" version="1805"><![CDATA[<img src="attachment.php?aid={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" />&nbsp;&nbsp;&nbsp;]]></template>
+		<template name="postbit_attachments_images_image" version="1805"><![CDATA[<img src="{$attachment_url}?aid={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" />&nbsp;&nbsp;&nbsp;]]></template>
```
Comment (euantorano) [inline install/resources/mybb_theme.xml]: Template version needs bumping:

```suggestion
		<template name="postbit_attachments_images_image" version="1826"><![CDATA[<img src="{$attachment_url}?aid={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" />&nbsp;&nbsp;&nbsp;]]></template>
```

File: install/resources/mybb_theme.xml [hunk 3]
```diff
@@ -9680,11 +9680,11 @@ if(use_xmlhttprequest == "1")
...
 		<template name="postbit_attachments_thumbnails" version="120"><![CDATA[<span class="smalltext"><strong>{$lang->postbit_attachments_thumbnails}</strong></span><br />
 {$post['thumblist']}
 <br />]]></template>
-		<template name="postbit_attachments_thumbnails_thumbnail" version="1805"><![CDATA[<a href="attachment.php?aid={$attachment['aid']}" target="_blank"><img src="attachment.php?thumbnail={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" /></a>&nbsp;&nbsp;&nbsp;]]></template>
+		<template name="postbit_attachments_thumbnails_thumbnail" version="1805"><![CDATA[<a href="attachment.php?aid={$attachment['aid']}" target="_blank"><img src="{$attachment_url}?thumbnail={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" /></a>&nbsp;&nbsp;&nbsp;]]></template>
```
Comment (euantorano) [inline install/resources/mybb_theme.xml]:
```suggestion
		<template name="postbit_attachments_thumbnails_thumbnail" version="1826"><![CDATA[<a href="attachment.php?aid={$attachment['aid']}" target="_blank"><img src="{$attachment_url}?thumbnail={$attachment['aid']}" class="attachment" alt="" title="{$lang->postbit_attachment_filename} {$attachment['filename']}&#13;{$lang->postbit_attachment_size} {$attachment['filesize']}&#13;{$attachdate}" /></a>&nbsp;&nbsp;&nbsp;]]></template>
```

File: install/resources/settings.xml (hunk 1 above)
Comment (euantorano) [reply install/resources/settings.xml]:
```suggestion
			<description><![CDATA[Select a default image to use for image attachments when users have no permission to download.]]></description>
```

File: install/resources/settings.xml [hunk 4]
```diff
@@ -1537,6 +1537,14 @@ min=1]]></optionscode>
 			<settingvalue><![CDATA[96]]></settingvalue>
 			<isdefault>1</isdefault>
 		</setting>
+		<setting name="attachimagestock">
+			<title>Stock Image Path</title>
```
Comment (euantorano) [inline install/resources/settings.xml]:
```suggestion
			<title>Default Image Path</title>
```

File: install/resources/settings.xml [hunk 5]
```diff
@@ -1537,6 +1537,14 @@ min=1]]></optionscode>
...
+			<description><![CDATA[Select an stock image to use for image attachments when users have no permission to download.]]></description>
+			<disporder>6</disporder>
+			<optionscode><![CDATA[text]]></optionscode>
+			<settingvalue><![CDATA[images/attachment_stock.svg]]></settingvalue>
```
Comment (euantorano) [inline install/resources/settings.xml:1545]: We should probably include a basic default stock image with the core.

**Review by euantorano — CHANGES_REQUESTED** [review]

**Comment by dvz** [comment]
This bug may be used as a feature - we could add a way to keep displaying original thumbnails (perhaps when the setting is empty).

**Comment by Sama34** [comment]
Sure, I will update this PR once we got the default image we are going to pack ( @justinsoltesz :P ).

File: install/resources/settings.xml (hunk 5 above)
Comment (Eldenroot) [reply install/resources/settings.xml:1545]: Also again rename attachment_stock into attachment_default :)

**Comment by dvz** [comment]
Looks like these are still unaddressed (https://github.com/mybb/mybb/issues/4240#issuecomment-775511385):
> - Permission handling for thumbnails need to be added to the attachment.php file.
> - Some users might decide to embed attachments directly using the [img] MyCode, this feature will also need to be reviewed.

**Comment by Eldenroot** [comment]
@Sama34 - please could you finish?! Thank you!

**Comment by Sama34** [comment]
We decided to postpone this for 1.9 @Eldenroot

**Comment by Eldenroot** [comment]
Not good, because this is quite annoying :(

**Pull request closed without merging**